csharp/src/Ice/Communicator-Properties.cs

```diff
@@ -52,6 +52,65 @@ internal PropertyValue(string v, bool u)
         public bool? GetPropertyAsBool(string name) =>
             GetPropertyAsInt(name) is int intValue ? intValue > 0 : (bool?)null;
 
+        public int? GetPropertyAsByteSize(string name)
+        {
+            lock (_properties)
+            {
+                if (_properties.TryGetValue(name, out PropertyValue? pv))
+                {
+                    pv.Used = true;
+
+                    if (pv.Val == "unlimited")
+                    {
+                        return 0;
+                    }
+
+                    // Match an integer followed letters
+                    Match match = Regex.Match(pv.Val, @"^([0-9]+[\.]?[0-9]*)([A-Z]+)$");
+
+                    if (!match.Success)
+                    {
+                        throw new InvalidConfigurationException(
+                            $"the value `{pv.Val}' of property `{name}' is not a byte size");
+                    }
+
+                    try
+                    {
+                        double value = double.Parse(match.Groups[1].Value, CultureInfo.InvariantCulture);
+                        string unit = match.Groups[2].Value;
+                        double size = unit switch
+                        {
+                            "B" => value,
+                            "KB" => 1024 * value,
+                            "MB" => 1024 * 1024 * value,
+                            "GB" => 1024 * 1024 * 1024 * value,
+                            _ => throw new FormatException($"unknown size unit `{unit}'"),
+                        };
+
+                        if (size > int.MaxValue)
+                        {
+                            throw new ArgumentOutOfRangeException($"the maximum size value is {int.MaxValue}B");
+                        }
+
+                        if (size <= 0)
```

Comment:
This check should be unnecessary as the regex won't allow negative values. Maybe just change it to an assert.

Reply:
I can change it to `== 0` , I don't think we want to accept `0KB` etc